### PR TITLE
Rename DeferredArtifact and remove DownloadCatalog

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #2968
https://pulp.plan.io/issues/2968

### CODE REVIEW

**Comment by jortel** [comment]
@dkliban originally named it `DeferredArtifact` so would like him to weigh in on that part at least.

File: plugin/pulpcore/plugin/download/factory.py [hunk 1]
```diff
@@ -36,7 +36,7 @@ def build(self, url, path=None, artifact=None):
         Args:
             url (str): The download URL.
             path (str): The optional absolute path to where the downloaded file is to be stored.
-            artifact (pulpcore.app.models.DeferredArtifact): An optional artifact.
+            artifact (pulpcore.app.models.RemoteArtifact): An optional artifact.
```
Comment (dkliban) [inline plugin/pulpcore/plugin/download/factory.py]: should probably rename this to remote_artifact

File: platform/pulpcore/app/models/content.py [hunk 2]
```diff
@@ -122,7 +122,7 @@ class Meta:
         unique_together = ('content', 'relative_path')
 
 
-class DeferredArtifact(Model):
+class RemoteArtifact(Model):
     """
     Stores information about an artifact that has not been retrieved yet.
```
Comment (jortel) [inline platform/pulpcore/app/models/content.py]: Probably want to reword the description to something like:
```
Represents an artifact that exists in a remote repository.
```

**Comment by daviddavis** [comment]
@dkliban @jortel addressed feedback and also tested against removals (which worked).

**Pull request merged**